flink-runtime/src/main/java/org/apache/flink/runtime/zookeeper/ZooKeeperStateHandleStore.java

```diff
@@ -154,54 +158,75 @@ public ZooKeeperStateHandleStore(
             throws PossibleInconsistentStateException, Exception {
         checkNotNull(pathInZooKeeper, "Path in ZooKeeper");
         checkNotNull(state, "State");
-
         final String path = normalizePath(pathInZooKeeper);
+        try {
+            try {
+                // Obtain the write-lock.
+                client.create().withMode(CreateMode.EPHEMERAL).forPath(getWriteLockPath(path));
```

Comment:
It could indeed result in the `NodeExistsException`. However this wouldn't introduce inconsistencies, because we don't discard the underlying state.

On the other hand, this would trigger the "unexpected" catch block bellow ... 🤔

Reply:
We can simply get rid of this write lock. It was meant just a safeguard, but it seems to introduce more corner cases. If we trust the leader election mechanism here, it's not needed. I'll remove it.